### PR TITLE
Initialize candidate directly instead of iterating the array of candidates

### PR DESCRIPTION
**Pull request by soulxu**
Using existing victimsMap to get the victims, then it is easy to build candidate
directly.

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 1]
```diff
@@ -330,13 +330,13 @@ func SelectCandidate(candidates []Candidate) Candidate {
 	victimsMap := candidatesToVictimsMap(candidates)
 	candidateNode := pickOneNodeForPreemption(victimsMap)
 
-	// Same as candidatesToVictimsMap, this logic is not applicable for out-of-tree
-	// preemption plugins that exercise different candidates on the same nominated node.
-	for _, candidate := range candidates {
-		if candidateNode == candidate.Name() {
-			return candidate
+	if victims, _ := victimsMap[candidateNode]; victims != nil {
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:333]: Thanks for the catch. Please keep the comments.

Comment (Huang-Wei) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:333]:
```suggestion
	if victims := victimsMap[candidateNode]; victims != nil {
```

Comment (soulxu) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:333]: Thanks, updated

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 2]
```diff
@@ -332,11 +332,13 @@ func SelectCandidate(candidates []Candidate) Candidate {
...
-			return candidate
+	if victims := victimsMap[candidateNode]; victims != nil {
+		return &candidate{
+			victims: victims,
+			name:    candidateNode,
 		}
 	}
+
 	// We shouldn't reach here.
 	klog.Errorf("None candidate can be picked from %v.", candidates)
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:343]:
```suggestion
	klog.Errorf("should not reach here, no candidate selected from %v.", candidates)
```

**Review by ahg-g — COMMENTED** [review]
/release-note-none
/kind cleanup
/priority backlog

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go (hunk 2 above)
Comment (soulxu) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go]: done

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 3]
```diff
@@ -332,13 +332,15 @@ func SelectCandidate(candidates []Candidate) Candidate {
...
+	if victims := victimsMap[candidateNode]; victims != nil {
+		return &candidate{
+			victims: victims,
+			name:    candidateNode,
 		}
 	}
+
 	// We shouldn't reach here.
-	klog.Errorf("None candidate can be picked from %v.", candidates)
+	klog.Errorf("Should not reach here, no candidate selected from %v.", candidates)
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:343]:
```suggestion
	klog.Errorf("should not reach here, no candidate selected from %v.", candidates)
```

**Review by ahg-g — COMMENTED** [review]
one nit, error messages starts with small letter

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go (hunk 3 above)
Comment (soulxu) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:343]: got it, thanks

**Pull request merged**